mm2src/lp_ordermatch/ordermatch_wasm_db.rs

```diff
@@ -0,0 +1,136 @@
+use async_trait::async_trait;
+use common::indexed_db::{DbIdentifier, DbInstance, DbUpgrader, IndexedDb, IndexedDbBuilder, OnUpgradeResult,
+                         TableSignature};
+use std::ops::Deref;
+use uuid::Uuid;
+
+pub use common::indexed_db::{cursor_prelude, DbTransactionError, DbTransactionResult, InitDbError, InitDbResult,
+                             ItemId};
+pub use tables::{MyActiveMakerOrdersTable, MyActiveTakerOrdersTable, MyFilteringHistoryOrdersTable,
+                 MyHistoryOrdersTable};
+
+const DB_NAME: &str = "ordermatch";
+const DB_VERSION: u32 = 1;
+
+pub struct OrdermatchDb {
+    inner: IndexedDb,
+}
+
+#[async_trait]
+impl DbInstance for OrdermatchDb {
+    fn db_name() -> &'static str { DB_NAME }
+
+    async fn init(db_id: DbIdentifier) -> InitDbResult<Self> {
+        let inner = IndexedDbBuilder::new(db_id)
+            .with_version(DB_VERSION)
+            .with_table::<MyActiveMakerOrdersTable>()
+            .with_table::<MyActiveTakerOrdersTable>()
+            .with_table::<MyHistoryOrdersTable>()
+            .with_table::<MyFilteringHistoryOrdersTable>()
+            .build()
+            .await?;
+        Ok(OrdermatchDb { inner })
+    }
+}
+
+impl Deref for OrdermatchDb {
+    type Target = IndexedDb;
+
+    fn deref(&self) -> &Self::Target { &self.inner }
+}
+
+pub mod tables {
+    use super::*;
+    use crate::mm2::lp_ordermatch::{MakerOrder, Order, TakerOrder};
+    use serde_json::Value as Json;
+
+    #[derive(Debug, Deserialize, Eq, PartialEq, Serialize)]
+    pub struct MyActiveMakerOrdersTable {
```

Comment:
I have an intuitive feeling that it might be worth splitting a Table and it's item into separate structs. They can be pinned to each other using generics or associated types of the TableSignature trait. It's just an idea, I'm unsure whether it's possible to do so without major refactoring. Maybe it doesn't make much sense in general :slightly_smiling_face:

Reply:
Actually, I didn't understand why it's needed :D
The only reason I see is the confusing names of the structures.

Reply:
> The only reason I see is the confusing names of the structures.

Actually yes, it's confusing because a specific struct represents table and table's item at the same time :slightly_smiling_face: